server/api/auth/permissions.py

```diff
@@ -0,0 +1,24 @@
+from rest_framework import permissions
+from rest_framework.request import HttpRequest
+
+
+class isStaffOrAuthenticated(permissions.BasePermission):
+    message = "User isn't staff! or isn't authenticated"
+
+    def has_permission(self, request: HttpRequest, view):
+        if request.user and request.user.is_staff:
+            return True
+        elif (
+            request.user.is_authenticated and request.method in permissions.SAFE_METHODS
+        ):
+            return True
+        return False
+
+
+class isStaffOrReadonly(permissions.BasePermission):
```

Comment:
nit: make these PascalCase

Reply:
The isAuthenticated classes, from my understanding, allow authenticated users to do POST, DELETE, PUT requests which should be allowed only for admins/staff

The isAdmin limits everything, I couldn't figure out a way to apply them to specific methods so I made my own permission classes

Reply:
Otherwise all good

Reply:
Doesn't really change anything. I will have to create a ReadOnly and an AuthenticatedReadOnly permissions and then combine them with the default permissions.